modal/_container_io_manager.py

```diff
@@ -196,10 +196,11 @@ async def _heartbeat_handle_cancellations(self) -> bool:
         return False
 
     @asynccontextmanager
-    async def heartbeats(self) -> AsyncGenerator[None, None]:
+    async def heartbeats(self, disable_init: bool) -> AsyncGenerator[None, None]:
```

Comment:
I think we want a different name here. It's sus that the name exists only very briefly before being 'thrown away'. Maybe `wait_for_mem_snap: bool`